### PR TITLE
Use polyhook with xmake package paradigm instead of custom cmakelists.txt

### PR DESCRIPTION
**Pull request by bitonality**
Remove the custom logic in polyhook package inclusion. This unifies xmake packages dir for stable caching on CI runners and polyhook will now respect the xmake package dir setting.

- Adds a package level setting for whether or not to use the zydis submodule that's included in the polyhook2 repo
- Adds package settings for shared/static runtimes
- Adds the ability to reuse the zydis package within polyhook (this also respects zydis AND zycore dependency trees and versioning)

### CODE REVIEW

**Review by localcc — APPROVED** [review]

**Comment by bitonality** [comment]
Heads up, I want testing on some more machines before merging so please don't merge!

**Comment by narknon** [comment]
Resulted in proxy being unable to load the ue4ss DLL

**Pull request closed without merging**
